### DISCUSSION TITLE
discussion: support for bootstrapping flux with other git providers

### DISCUSSION THREAD

**Discussion by jtogden** [onedr0p/cluster-template#1579] [Ideas]
@onedr0p love your work here.
Before I put in the work on a pull request, would you be willing to discuss supporting bootstrapping other git providers other than GitHub.  My use case is using this with my home lab and GitLab.com, but I could see this being used with private git servers as well.

**Comment by onedr0p** [comment]
I see the appeal however there's not only GitHub with flux but also useful GitHub workflows that have been put together. So sure the flux parts and bootstrapping could be easily added for supporting gitlab (or any other provider) but there're a lot of other parts that wouldn't easily translate over and would take a massive refactor to support.

Overall having less options means the current logic can remain lean and easy to grok, I feel like layering on many features or knobs for users can make this project harder to follow for users new to k8s.

**Comment by jtogden** [comment]
I completely respect the desire to keep opinionated platforms simple.

I may still borrow (giving this project credit of course) several aspects of this and create a GitLab opinionated version.

**Comment by onedr0p** [comment]
If you do feel free to link it here and I'll drop a link on the README.
